pgx/src/datum/date.rs

```diff
@@ -90,7 +90,9 @@ impl Date {
 
     #[inline]
     pub fn to_posix_time(&self) -> libc::time_t {
-        libc::time_t::from(self.to_unix_epoch_days()) * libc::time_t::from(pg_sys::SECS_PER_DAY)
+        let secs_per_day: libc::time_t =
+            pg_sys::SECS_PER_DAY.try_into().expect("couldn't fit time into time_t");
+        libc::time_t::from(self.to_unix_epoch_days()) * secs_per_day
```

Comment:
Technically, whether a system uses a 32-bit or 64-bit value for `time_t` is actually only incidentally historically related. Sometimes a 32-bit system will use a 64-bit `time_t`. Sometimes a 64-bit system will use a 32-bit `time_t`. Sometimes `time_t` is an unsigned integer. Sometimes it is a signed integer.

I don't think this actually affects this code's current correctness, it just felt worthy of note.